File: pset6/sentimental-readability/readability.py
# TODO

from cs50 import get_string


def main():
    Text = get_string("Text: ")
    # letters
    l = count_letters(Text)
    # words
    w = count_words(Text)
    # sentences
    s = count_sentences(Text)
    # putting it all together
    L = (l * 100) / (w + 1)
    S = (s * 100) / (w + 1)
    Grade = round(0.0588 * L - 0.296 * S - 15.8)

    # if the grade number is less than 1
    if Grade < 1:
        print("Before Grade 1")
    # if the grade number is 16 of higher
    elif Grade >= 16:
        print("Grade 16+")
    # if the grade number is between 1 and 16, round the grade number to the nearest int
    else:
        print(f"Grade {Grade}")


# count letters
def count_letters(Text):
    letters = 0
    for i in range(len(Text)):
        if (Text[i].isalpha()):
            letters += 1
    return letters


# count words
def count_words(Text):
    word = Text.count(" ")
    return word


# count sentences
def count_sentences(Text):

    aa = Text.count('.')
    cc = Text.count('!')
    bb = Text.count('?')

    sentences = int(aa + cc + bb)
    return sentences


main()

File: pset2/readability/readability.c

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string Text);
int count_words(string Text);
int count_sentences(string Text);

int main(void)
{
    string Text = get_string("Text: ");
    //letters
    float l = count_letters(Text);
    //words
    float w = count_words(Text);
    //sentences
    float s = count_sentences(Text);
    //Putting it All Together
    double L = (l * 100) / (w + 1);
    double S = (s * 100) / (w + 1);
    double Grade = 0.0588 * L - 0.296 * S - 15.8;

    //If the Grade number is less than 1
    if (Grade < 1)
    {
        printf("Before Grade 1\n");
    }
    //If the resulting Grade number is 16 or higher
    else if (Grade >= 16)
    {
        printf("Grade 16+\n");
    }
    //if the resulting is between 1 and 16, round the resulting Grade number to the nearest int!
    else
    {
        printf("Grade %i\n", (int) round(Grade));
    }
}


//count letters
int count_letters(string Text)
{

    int letters = 0;
    for (int i = 0; i < strlen(Text); i++)
    {
        if ((isupper(Text[i])) || (islower(Text[i])))
        {
            letters++;
        }
    }
    return letters;
}

//count words
int count_words(string Text)
{
    int words = 0;
    for (int i = 0; i < strlen(Text); i++)
    {
        if (Text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

//count sentences
int count_sentences(string Text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(Text); i++)
    {
        if (Text[i] == '.' || Text[i] == '!' || Text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}